src/ripple/app/rdb/backend/RelationalDBInterfaceSqlite.cpp

```diff
@@ -1175,32 +1351,38 @@ RelationalDBInterfaceSqliteImp::oldestAccountTxPage(
         return {ret, newmarker};
     }
 
-    /* else use shard databases */
-    AccountTxPageOptions opt = options;
-    int limit_used = 0;
-    iterateTransactionForward(
-        opt.minLedger ? seqToShardIndex(opt.minLedger)
-                      : std::optional<std::uint32_t>(),
-        [&](soci::session& session, std::uint32_t shardIndex) {
-            if (opt.maxLedger != UINT32_MAX &&
-                shardIndex > seqToShardIndex(opt.minLedger))
-                return false;
-            auto [marker, total] = ripple::oldestAccountTxPage(
-                session,
-                idCache,
-                onUnsavedLedger,
-                onTransaction,
-                opt,
-                limit_used,
-                page_length);
-            opt.marker = marker;
-            if (total < 0)
-                return false;
-            limit_used += total;
-            return true;
-        });
+    /* else use shard databases, if available */
+    if (shardStoreExists() && useTxTables_)
```

Comment:
I think all of these functions can just return early if `useTxTables_` is false. There's no need to do anything else. So at the top of the function:

```
if(!useTxTables_)
    return {};
```

Reply:
Done